### ISSUE TITLE
Example of using DynTemplate

### ISSUE THREAD

**Issue by garytaylor** [djc/askama#713]
Being relatively new to rust, as I am sure others are, I am struggling to have a struct that derives 'Template' contain a field of type 'Anything that implements Template'.  In other words, I want to dynamically render another template inside mine (not just include some html - I actually want to render a completely new template)

It seems I cannot do anything like put a box around Template as it is not object safe - the compiler suggested using DynTemplate

However, I cannot figure out how to use it

Maybe an example could be added to show this ?

Thanks

Gary

**Comment by djc** [comment]
For a template within a template, you can just treat the inner value as an expression in the outer template. Because all implementers of Template also implement fmt::Display, that should just work.

**Comment by garytaylor** [comment]
Yes, but I want to pass in the inner template as opposed to it being hard coded.  I have developed a "Model" component that can contain any other type of component - but the inner component needs to be passed in somehow - does that make sense ?

**Comment by djc** [comment]
Sure, it could also just be a `T: Template` or `T: fmt::Display`. `DynTemplate` is implemented by the macro for any type that impls `Template` and can be used like any object-safe trait.

**Comment by garytaylor** [comment]
I have yet to get used to generics - it's my next challenge in learning rust.  The whole project I am doing is really just for learning, but it will have a final use if it is good enough quality.
I come from a ruby background but also java from 20 years ago.  I will take a look at going down that path - thanks for your help

**Comment by Kijewski** [comment]
@RoastBeefer00, you have to disable the automatic escaping: <https://rinja-rs.github.io/askama/filters.html#safe>.

**Comment by RoastBeefer00** [comment]
Thank you @Kijewski, that worked!

`{{ page|safe }}` renders the inner template properly :)

**Comment by norbertkeri** [comment]
Is there a way to do this but allow DynTemplate to have a lifetime? Eg. I'm trying the following:

```rust
#[derive(Template)]
#[template(path = "base.html", print = "all")]
pub struct BaseTemplate<'a> {
    pub main: Box<dyn DynTemplate + 'a>,
}
```

But unfortunately this does not compile:

```
error[E0521]: borrowed data escapes outside of method
 --> backend/src/template/mod.rs:4:10
  |
4 | #[derive(Template)]
  |          ^^^^^^^^
  |          |
  |          `self` is a reference that is only valid in the method body
  |          `self` escapes the method body here
  |          lifetime `'a` defined here
  |          argument requires that `'a` must outlive `'static`
```

(I realize this looks very similar to what can be accomplished with {% extends ... %} but I can't use that for other reasons)
